src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -0,0 +1,70 @@
+package wooteco.subway.service;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineRequest;
+import wooteco.subway.dto.LineResponse;
+import wooteco.subway.error.exception.NotFoundException;
+
+@Transactional(readOnly = true)
+@Service
+public class LineService {
+
+    private final LineDao lineDao;
+
+    public LineService(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    @Transactional
+    public LineResponse save(LineRequest lineRequest) {
+        if (lineDao.existsByName(lineRequest.getName())) {
+            throw new IllegalArgumentException(lineRequest.getName() + "은 이미 존재하는 노선 이름입니다.");
+        }
+
+        Line line = lineDao.save(new Line(lineRequest.getName(), lineRequest.getColor()));
+        return new LineResponse(line);
+    }
+
+    public LineResponse findById(Long id) {
+        Line line = getLine(id);
+        return new LineResponse(line);
+    }
+
+    public List<LineResponse> findAll() {
+        return lineDao.findAll()
+                .stream()
+                .map(LineResponse::new)
+                .collect(toList());
+    }
+
+    @Transactional
+    public LineResponse update(Long id, LineRequest lineRequest) {
+        getLine(id);
+
+        try {
+            lineDao.update(id, new Line(lineRequest.getName(), lineRequest.getColor()));
+        } catch (DuplicateKeyException e) {
+            throw new IllegalArgumentException(lineRequest.getName() + "은 이미 존재하는 노선 이름입니다.");
+        }
```

Comment:
이쪽은 애플리케이션에서 검사하지 않으시는 이유가 있을까요?

```suggestion
        validateName(lineRequest)
        lineDao.update(id, new Line(lineRequest.getName(), lineRequest.getColor()));
```

저장과 비즈니스흐름이 다르네요 👀